### ISSUE TITLE
Improve error message when trying to connect to chat using an app access token

### ISSUE THREAD

**Issue by evandikt** [d-fischer/twitch#114]
HI,

it's possible to connect to an irc with [oauth app token](https://dev.twitch.tv/docs/authentication/getting-tokens-oauth/#oauth-client-credentials-flow) ?

Actually i got a 
```
[2020-03-09T09:01:46.054Z] ❌ ircv3 ERROR Disconnected unexpectedly: [1006] 
(node:7325) UnhandledPromiseRejectionWarning: Error: required parameter "nick" not found in command "NICK"
```
And i can't pass it as a param in `TwitchClient.withCredentials(process.env.TWITCH_CLIENT_ID, token.access_token)` :(

Thx

**Comment by d-fischer** [comment]
No, because app tokens are not associated with a user. Apps can not be in chat, only users can. You need to create a Twitch account for your bot with the name you want the bot to have, then get an access token for that user, in order to connect to chat.

Still leaving this open because that's definitely not the kind of error message that should appear in that case.

**Comment by d-fischer** [comment]
Released in 4.1.0-pre.10.

**Comment by Cobertos** [comment]
Hmm, I seem to still be getting the error mentioned above.

```javascript
{
    "twitch": "^4.2.7",
    "twitch-auth": "4.2.7",
    "twitch-chat-client": "4.2.7",
}
```

![image](https://user-images.githubusercontent.com/6759176/96994026-16184180-14fa-11eb-8886-54b860321fad.png)
